Travis ci

lens0021 opened the pull request:
issue: https://github.com/femiwiki/mediawiki/issues/242

lens0021 committed:
Introduce travis-ci

1.  Create '.travis.yml'
2.  Update gitignore and dockerignore files
3.  Create composer.json and composer.lock
4.  Import .phpcs.xml from https://github.com/wikimedia/mediawiki/blob/REL1_31/.phpcs.xml
5.  Update editorconfig for composer.lock and phpcs.xml

lens0021 committed:
Update README.md

1.  Add a badge of Travis CI
2.  Mention 'composer test'
3.  Organize instructions

simnalamburt commented on composer.json:
```diff
@@ -0,0 +1,18 @@
+    "mediawiki/mediawiki-codesniffer": "23.0.0",
+    "phpunit/phpunit": "4.8.36 || ^6.5",
+    "jakub-onderka/php-parallel-lint": "0.9.2",
+    "mediawiki/minus-x": "0.3.1"
```
이거 하지맙시다

lens0021 replied:
3-6인가로 표시되어 있는데 line 6만 얘기하신 거 같아서 minus-x만 뺐습니당

simnalamburt commented on .gitignore:
```diff
@@ -1,3 +1,10 @@
+/composer.lock
+
+# PHP CodeSniffer
+.phpcs.xml
```
이거 gitignore에 넣지말고 PHP lint 할거면 `.phpcs.xml` 파일을 저장소에 커밋해버리는게 나을듯

simnalamburt replied:
.gitignore에서 생긴 변화들 .dockerignore에도 똑같이 반영해주세요

simnalamburt commented on .gitignore:
```diff
@@ -1,3 +1,10 @@
+
+# Composer
+/vendor
+/composer.lock
```
이것도 무시하지말고 커밋하는게 나을듯

simnalamburt commented on .travis.yml:
```diff
@@ -0,0 +1,176 @@
+            php composer-setup.php --install-dir=/usr/local/bin --filename=composer --quiet
+        - docker exec $(docker container ls -qf name=mediawiki_fastcgi)
+            composer install --prefer-source --quiet --no-interaction
+        - "docker exec $(docker container ls -qf name=mediawiki_fastcgi)
```
남의 라이브러리들까지 우리가 테스트할필요는 없을듯

simnalamburt commented on .travis.yml:
```diff
@@ -0,0 +1,176 @@
+        - docker exec $(docker container ls -qf name=mediawiki_fastcgi)
+            php composer-setup.php --install-dir=/usr/local/bin --filename=composer --quiet
+        - docker exec $(docker container ls -qf name=mediawiki_fastcgi)
+            composer install --prefer-source --quiet --no-interaction
```
아래와 같이 하면 docker exec 한번에 여러 줄 실행할 수 있어요

```
docker exec -it blabla bash -c 'echo hi && echo hi;'
```

외따옴표 안에 커맨드 길게 쓰는게 불편하면

```
xargs -0 docker exec -it blabla bash -c -- <<'EOF'
echo hi &&
echo hi
EOF
```

이렇게하시면 됨

simnalamburt reviewed (changes requested):
`.travis.yml` 안에있는 긴 셸스크립트들 읽기힘든데 주석좀 넣어주세요 특히 반복문 근처 코드들

simnalamburt commented on .travis.yml:
```diff
@@ -0,0 +1,176 @@
+        - curl -Lo .phpcs.xml https://raw.githubusercontent.com/wikimedia/mediawiki/${MEDIAWIKI_BRANCH}/.phpcs.xml
+      script:
+        - php -l configs/LocalSettings.php
+        - php -l configs/secret.example.php
```
`secret.php.example`로 고쳐야함

simnalamburt commented on .travis.yml:
```diff
@@ -0,0 +1,176 @@
+        - git clone https://github.com/femiwiki/database.git ~/swarm --depth=1
+      before_script:
+        - cp configs/env.example configs/env
+        - cp configs/secret.example.php configs/secret.php
```
`secret.php.example`로 고쳐야함

lens0021 committed:
Update travis.yml: Multiple improvements

1.  Don't lint 'secret.php'
2.  Preperation steps are 'before_script' phase not 'install' phase
3.  Reorder some steps
4.  Don't wait until 'mediawiki_parsoid' and 'mediawiki_fastcgi' are up
5.  Add comments

lens0021 committed:
Update travis.yml: Don't do phpunit and minus-x

It is too much to test the external codes

simnalamburt committed:
Update phpcs.xml

1.  Made it not to ignore LocalSettings.php
2.  Removed redundant rules

simnalamburt committed:
Update travis.yml: Speed up

1.  Simplify multiple codes
2.  Remove 'sleep 60'

simnalamburt committed:
Update composer.json: Remove unused dependencies

simnalamburt committed:
Lint LocalSettings.php

simnalamburt committed:
Update composer scripts
